cluster-autoscaler/core/scaledown/eligibility/eligibility_test.go

```diff
@@ -50,18 +50,45 @@ func TestFilterOutUnremovable(t *testing.T) {
 	noScaleDownNode.Annotations = map[string]string{ScaleDownDisabledKey: "true"}
 	SetNodeReadyState(noScaleDownNode, true, time.Time{})
 
+	unreadyNode := BuildTestNode("unready", 1000, 10)
+	SetNodeReadyState(unreadyNode, false, time.Time{})
+
 	bigPod := BuildTestPod("bigPod", 600, 0)
 	bigPod.Spec.NodeName = "regular"
 
 	smallPod := BuildTestPod("smallPod", 100, 0)
 	smallPod.Spec.NodeName = "regular"
 
-	testCases := []struct {
+	type testCase struct {
 		desc  string
 		nodes []*apiv1.Node
 		pods  []*apiv1.Pod
 		want  []string
-	}{
+	}
+
+	runTestCase := func(tc testCase, scaleDownUnready bool) {
```

Comment:
nit: Why isn't `scaleDownUnready` a part of `testCase`?

Reply:
I created a fixup to make this part of `testCase`. The diff is a bit ugly (due to go indentation). Shall I *force-push*?

Reply:
Yeah, please do!

Reply:
@x13n done